<?xml version="1.0"?>
<launch>
    
    <include file="$(find realsense2_camera)/launch/rs_camera.launch">
        <arg name="color_width" value="640" />
        <arg name="color_height" value="480" />
        <arg name="filters" value="pointcloud" />
        <arg name="enable_pointcloud" default="true"/>
        <arg name="gyro_fps" default="200"/>
        <arg name="accel_fps" default="100"/>
        <arg name="enable_gyro" default="true"/>
        <arg name="enable_accel" default="true"/>
        <arg name="publish_tf" default="true"/>
        <arg name="unite_imu_method" default="linear_interpolation"/>
    </include>

    <param name="scan_period" value="0.1" />
    <param name="vertical_angle" type="double" value="2.0" />
    <param name="max_dis" type="double" value="9.0" />
    <param name="map_resolution" type="double" value="0.05" />
    <param name="min_dis" type="double" value="0.2" />

    <node pkg="ssl_slam" type="ssl_slam_laser_processing_node" name="ssl_slam_laser_processing_node" output="log"/>
    <node pkg="ssl_slam" type="ssl_slam_odom_estimation_node" name="ssl_slam_odom_estimation_node" output="log"/>
    <node pkg="ssl_slam" type="ssl_slam_laser_mapping_node" name="ssl_slam_laser_mapping_node" output="log"/>

    <arg name="rviz" default="true" />
    <group if="$(arg rviz)">
        <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find ssl_slam)/rviz/ssl_slam_mapping.rviz" />
    </group>


</launch>
